Improved testing for nohup, #173

riczorn opened the pull request:
### Description:

Changed `nohup` detection to avoid using which, both in Tasks and in RequiredExecutablesCheck

riczorn committed:
Improved testing for nohup, #173

AltamashShaikh commented on Diagnostic/RequiredExecutablesCheck.php:
```diff
@@ -65,6 +65,6 @@ public function execute()
 
     private function isNohupPresent()
     {
-        return (bool) @shell_exec('which nohup');
```
@riczorn What is the actual benefit of using `which nohup` vs `nohup --version` here ?

riczorn replied:
`which nohup` returns the nohup path, but it does require the `which` executable which may not be present, so it may fail even though nohup is present. `nohup --version` does not rely on it.

AltamashShaikh commented on Diagnostic/RequiredExecutablesCheck.php:
```diff
@@ -65,6 +65,6 @@ public function execute()
 
     private function isNohupPresent()
```
Also you can change this method access to public and use the same one in Tasks.php
```suggestion
    public function isNohupPresent()
```

AltamashShaikh replied:
@riczorn You changed the identifier back to private

riczorn replied:
But you had changed it and I didn't touch it! I added a commit with it public again do you see it now?

AltamashShaikh replied:
I can see it now

AltamashShaikh commented on Tasks.php:
```diff
@@ -344,7 +344,11 @@ private static function getNohupCommandIfPresent()
-
-        return 'nohup';
+        
+        if (!empty(@shell_exec('nohup --version'))) {
```
```suggestion
$requiredExecutablesCheck = StaticContainer::get(RequiredExecutablesCheck::class);

        if ($requiredExecutablesCheck->isNohupPresent()) {
            return 'nohup';
        } else {
            return '';
        }
```

You need to add this use statement too

`use Piwik\Plugins\GoogleAnalyticsImporter\Diagnostic\RequiredExecutablesCheck;`

AltamashShaikh reviewed (changes requested):
@riczorn left few suggestions

riczorn committed:
Update Tasks.php

Co-authored-by: Altamash Shaikh <altu9594@gmail.com>

AltamashShaikh commented on Tasks.php:
```diff
@@ -344,7 +344,17 @@ private static function getNohupCommandIfPresent()
+$requiredExecutablesCheck = StaticContainer::get(RequiredExecutablesCheck::class);
 
-        return 'nohup';
+        if ($requiredExecutablesCheck->isNohupPresent()) {
```
@riczorn looks like you messed up the lines while applying the suggestion please fix it

AltamashShaikh reviewed (changes requested):
@riczorn Left 1 comment

riczorn committed:
typos fixed

sorry I just pressed "Accept suggestion" in github and apparently it didn't quite work out, this should do.

riczorn committed:
identifier back to public

AltamashShaikh reviewed (approved):
Looks good, @snake14 can you also check once

snake14 reviewed (approved):
The changes look good to me. When I applied them to my local environment and ran an import, I saw the nohup output as expected.

snake14 commented:
The failing builds don't appear to be related to the code changes, so I'm going to go ahead and merge.
